okhttp/src/main/java/okhttp3/Handshake.kt

```diff
@@ -183,8 +186,8 @@ class Handshake private constructor(
       peerCertificates: List<Certificate>,
       localCertificates: List<Certificate>
     ): Handshake {
-      return Handshake(tlsVersion, cipherSuite, peerCertificates.toImmutableList(),
-          localCertificates.toImmutableList())
+      return Handshake(tlsVersion, cipherSuite, localCertificates.toImmutableList(),
+          { peerCertificates.toImmutableList() })
```

Comment:
you gotta call toImmutableList() early, otherwise mutating the list after the call impacts the Handshake

Reply:
Good catch